packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -730,6 +764,51 @@ class _TextSelectionGestureDetectorState extends State<TextSelectionGestureDetec
     }
   }
 
+  DragStartDetails _lastDragStartDetails;
+
+  void _handleDragStart(DragStartDetails details) {
+    assert(_lastDragStartDetails == null);
+    _lastDragStartDetails = details;
+    if (widget.onDragSelectionStart != null) {
+      widget.onDragSelectionStart(details);
+    }
+  }
+
+  Timer _dragUpdateTimer;
+  DragUpdateDetails _lastDragUpdateDetails;
+
+  void _handleDragUpdate(DragUpdateDetails details) {
+    _lastDragUpdateDetails = details;
+    // Only schedule a new timer if there's no one pending.
+    _dragUpdateTimer ??= Timer(_kDragSelectionUpdateThrottle, _handleDragUpdateThrottled);
+  }
+
+  void _handleDragUpdateThrottled() {
+    assert(_lastDragStartDetails != null);
+    assert(_lastDragUpdateDetails != null);
```

Comment:
check that the State is still in the tree when this happens.

Reply:
Good catch. I'll override the `dispose` method and cancel the timer there instead. Does that sound good?